src/switchImplIntf.ml

```diff
@@ -0,0 +1,60 @@
+open Import
+
+module Lsp : sig
+  val switch : LanguageClient.t -> TextDocument.t -> string option Promise.t
+end = struct
+  let switch client document =
+    let targetFileName : string option Promise.t =
+      LanguageClient.sendRequest client ~meth:"ocaml/didSwitchImplIntf"
```

Comment:
Let's call this `ocamllsp/` rather than `ocaml/`. It's not only for ocaml, and it's server specific so we should name it after the server.